sat-plugin/src/main/java/org/openhab/tools/analysis/tools/CheckstyleChecker.java

```diff
@@ -127,4 +174,40 @@ public void execute() throws MojoExecutionException {
 
     }
 
+    private void generateRulesXmlFile(String location) throws MojoExecutionException {
+        try {
+            String rulesLocation = getLocation(checkstyleRuleset, DEFAULT_RULE_SET_XML);
```

Comment:
Are we using this method even if we do not skip anything, if I remember correctly XML manipulation like this is rather expensive, so please skip it if its not needed.

Reply:
We can do this for pmd, but we can't for checkstyle, because the added types to the rules are not part of the official API. We have to at least remove them before executing.

Reply:
What is the (performance) impact of this?